gui/src/ts/components/pages/NodePage.tsx

```diff
@@ -109,30 +85,55 @@ export const NodePage: React.FunctionComponent = () => {
     } else if (!edge.predicate) {
       continue;
     }
-    const edges = subjectOfEdgesByPredicate[edge.predicate];
+    const edges = predicateSubjects[edge.predicate];
```

Comment:
lodash has an `update` for this idiom (~ Python `setdefault`). You can also use space-lift if you're feeling adventurous.